Adds filtering by repo

ZooeyMiller opened the pull request:
Using rummage and creating our own form with an autocomplete you can now filter the issues by a search term (which can be autocompleted with a name of a repo).

fix #220

ZooeyMiller committed:
adds search for repo
#220

ZooeyMiller committed:
adds search functionality
using a mix of rummage and some elixir-fu we have a search with an autocomplete/drop down
#220

ZooeyMiller commented:
Travis is failing, but tests and coverage working fine on local machine, looking to see if updating travis to 1.5.1 (same as local) resolves.

ZooeyMiller committed:
updates travis version
https://github.com/dwyl/tudo/pull/223#issuecomment-328120810

ZooeyMiller committed:
explicitly passes in params in text
https://github.com/dwyl/tudo/pull/223#issuecomment-328120810 #220

ZooeyMiller committed:
wraps get call in brackets in hope that function is passed in properly
https://github.com/dwyl/tudo/pull/223#issuecomment-328120810 #220

ZooeyMiller committed:
updates elixir version with offending functin wrapped in brackets
https://github.com/dwyl/tudo/pull/223#issuecomment-328120810 #220

ZooeyMiller committed:
replaces pattern match in case with default
#220 #224

samhstn commented on web/router.ex:
```diff
@@ -15,6 +15,7 @@ defmodule Tudo.Router do
     pipe_through :browser # Use the default browser stack
 
     get "/", PageController, :index
+    post "/", PageController, :search
```
I think this should still be a `get` request if you want to ensure the http verbs keep semantic meaning.
Perhaps look at using query params in the `index` or a different `search` controller

samhstn commented on web/templates/page/index.html.eex:
```diff
@@ -6,7 +6,27 @@
+  <% end %>
+  <datalist id="repos">
+    <%= for repo_name <- @repos do %>
+    <option value= <%= repo_name %> />
```
Should the option value [be in a string and have `innerHTML`](https://www.w3schools.com/tags/tag_option.asp)?

```elixir
<option value="<%= repo_name %>"><%= String.capitalize repo_name %></option>
```

ZooeyMiller replied:
That would be for a `select` tag, where this is a `datalist` to act as an autocomplete for a text input.

samhstn replied:
I see, haven't really used `datalist`. It seems that [it isn't supported by safari](https://www.w3schools.com/tags/tag_datalist.asp), could this be a problem?

ZooeyMiller replied:
It's true, it's not supported by safari, for the functionality we wanted (a search function with a drop down) datalist is the main way to do it (as far as I know). Safari users will still be able to use the search function, they just won't get the autocomplete. 

The only other way to achieve this could be front end javascript, which so far the site has none of.

samhstn replied:
I think autocomplete should be implemented with javascript, if there isn't any yet, then look at adding some (it's impressive that you've gone about autocomplete it without though).

I think that excluding safari users will effect a much larger portion of people than those who don't have javascript turned on.

ZooeyMiller replied:
For now, so that we can get this merged in, perhaps we remove the autocomplete feature, just leaving a search, and work on autocomplete as a separate issue/PR? 

Just as it isn't _necessary_ for searching by repo to have the autocomplete, and it will take a bit of extra time to implement the autocomplete with front end js.

samhstn commented on web/templates/page/index.html.eex:
```diff
@@ -6,7 +6,27 @@
+    <%= for repo_name <- @repos do %>
+    <option value= <%= repo_name %> />
+      <% end %>
+    </datalist>
```
Tidying up the indenting here would also probably be good

samhstn reviewed (changes requested):
Sorry, this took so long.

Let me know what you think about the minor points, then happy to merge

ZooeyMiller committed:
change search to get request
also fixes indentation #223 https://github.com/dwyl/tudo/pull/223#pullrequestreview-61858983

ZooeyMiller committed:
merge master into branch

samhstn reviewed (approved):
Fair enough, looks good 👍 

Happy to merge when merge conflicts are fixed

Jbarget reviewed (approved):
👍
